docs/versions/index.md

```diff
@@ -0,0 +1,124 @@
+---
+navigation_title: "Versions and types"
+---
+# Documenting versions and deployment types
+
+In the new documentation system, we document features in a centralized place, regardless of the software version or deployment type it applies to. 
+For example, we might document the serverless and v9 implementations of a feature on a single page, and then use content patterns to highlight where prerequisites, options, or behaviors differ between these implementations.
+
+This approach improves our documentation in several ways: 
+
+* When a user arrives in our documentation from an outside source, they'll be likelier to land on a page or section that applies to them.
+* There is a single "source of truth" for each feature, which helps us to maintain consistency, accuracy, and maintainability of our documentation over time, and avoids "drift" between multiple similar sets of documentation.
+* Comparing and contrasting differences helps users to understand what is available to them, and improves awareness of the ways certain offerings might improve their experience.
+
+::::{note}
+This page documents the first version of our approach to documenting differences in versions and deployment types, using our current technologies. 
+Our approach might change as additional documentation features become available.
+::::
+
+## Basic concepts and principles
+
+* [Versioning facets](#versioning-facets)
+* [Defaults and hierarchy](#defaults-and-hierarchy)
+* [Versions and lifecycle states](#versions-and-lifecycle-states)
+
+### Versioning facets
+There are multiple facets to versioning that we need to consider: 
+
+* **Elasticsearch / Kibana flavor:** The feature base used for basic functionality. Either **Serverless** (sometimes "Elastic Cloud Serverless") or **Stack <version>**
```

Comment:
Rather than limiting this to Elasticsearch and Kibana (since other things vary by deployment type too), can we generalize to something like this?
```suggestion
* **Platform or product versions:** For example, Elastic Stack 9.0.0. This is facet is currently not applicable in the context of Elastic Cloud Serverless.
```